scripts/provisioning/upgradecmd.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+from setupcmd import SetupCmd, S3PROVError
+from merge import merge_configs
+import os
+import shutil
+
+class UpgradeCmd(SetupCmd):
+
+  """Upgrade Setup Cmd."""
+  name = "upgrade"
+
+  def __init__(self, config: str, services: str, commit: str = None):
+    """Constructor."""
+    try:
+      super(UpgradeCmd, self).__init__(config, services, commit)
+    except Exception as e:
+      raise e
+
+  def process(self):
+    """Main processing function."""
+    self.logger.info(f"Processing phase = {self.name}, config = {self.url}, service = {self.services}, commit = {self.commit}")
+    try:
```

Comment:
Add #TODO - Print the existing version and upgrade version
Existing version will come from confstore.
Upgrade version will come from release.info file or rpm.
(for s3).

Reply:
Done